streams/src/test/java/org/apache/kafka/streams/tests/EosTestDriver.java

```diff
@@ -68,72 +69,115 @@ private static synchronized void updateNumRecordsProduces(final int delta) {
     }
 
     static void generate(final String kafka) {
+        try {
```

Comment:
I think the `Exit.addShutdownHook` block can be outside of the largest `try/finally` block of the method.

Reply:
Maybe. Does it matter?

Reply:
I should added in my previous comment it's a `nit`.